### PR TITLE
Usb Cam Sensor FPGA driver buffer count is set to four and add set-trigger function.

### PR DESCRIPTION
**Pull request by cicelprg**
Usb Cam Sensor FPGA driver buffer count is set to four and add set-trigger function.

### CODE REVIEW

File: third_party/adv_plat/include/adv_trigger_ctl.h [hunk 1]
```diff
@@ -0,0 +1,82 @@
+/************************************************************************
+*
+* Copyright (c) 2016 Baidu.com, Inc. All Rights Reserved
```
Comment (startcode) [inline third_party/adv_plat/include/adv_trigger_ctl.h]: use APACHE 2.0 License

File: third_party/adv_plat/include/adv_trigger_ctl.h [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+int adv_trigger_disable(const char *video_path);
+int adv_trigger_delay_ctl(const char *video_path, int action,
+		unsigned int *trigger_delay, unsigned int *exp_time);
+
+#define	FLAG_GPS_VALID		(1 << 0)
+#define	FLAG_PPS_VALID		(1 << 1)
+
+/* FPGA board trigger status. */
+struct adv_zynq_status {
+	struct {
```
Comment (startcode) [inline third_party/adv_plat/include/adv_trigger_ctl.h]: please make sure your code follows google style.

File: third_party/adv_plat/include/bcan_defs.h [hunk 3]
```diff
@@ -0,0 +1,56 @@
+/************************************************************************
+*
+* Copyright (c) 2016 Baidu.com, Inc. All Rights Reserved
```
Comment (startcode) [inline third_party/adv_plat/include/bcan_defs.h]: use APACHE 2.0 license

File: third_party/adv_plat/include/bcan_defs.h [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+*
+************************************************************************/
+
+/**
+* @file bcan_defs.h
+*
+* Baidu ADV (Autonomous Driving Vehicle) bcan basic definitions.
+**/
+
+#ifndef BCAN_DEFS_H
```
Comment (startcode) [inline third_party/adv_plat/include/bcan_defs.h]: please use the same convention as other Apollo code files
should be #ifndef THIRD_PARTY_ADV_PLAT_INCLUDE_BCAN_DEFS_H_
also make this change to all other header files

File: third_party/adv_plat/include/bcan_defs.h [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+#include <linux/time.h>
+#endif
+
+#define	BCAN_EXTENDED_FRAME	0x20000000
+
+/*
+ * Baidu CAN message definition
+ */
+typedef struct bcan_msg {
+	unsigned int	bcan_msg_id; /* source CAN node id */
```
Comment (startcode) [inline third_party/adv_plat/include/bcan_defs.h]: use two space indents

File: third_party/adv_plat/include/bcan_defs.h [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	unsigned char	bcan_msg_rsv[3];
+	unsigned char	bcan_msg_data[8]; /* message data */
+	struct timeval	bcan_msg_timestamp;
+} bcan_msg_t;
+
+
+/*
+ * CAN error code
+ */
+enum bcan_err_code {
```
Comment (startcode) [inline third_party/adv_plat/include/bcan_defs.h]: use CamelCase for type name

File: third_party/adv_plat/include/bcan_defs.h [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+#else
+#include <linux/time.h>
+#endif
+
+#define	BCAN_EXTENDED_FRAME	0x20000000
+
+/*
+ * Baidu CAN message definition
+ */
+typedef struct bcan_msg {
```
Comment (startcode) [inline third_party/adv_plat/include/bcan_defs.h]: use CamelCase for typename

File: third_party/adv_plat/include/zynq_api.h [hunk 8]
```diff
@@ -0,0 +1,470 @@
+/*
+ * zynq_api.h: Zynq driver APIs(exported ioctls and structures)
+ *
```
Comment (startcode) [inline third_party/adv_plat/include/zynq_api.h]: Use proper License

File: third_party/adv_plat/include/bcan_defs.h (hunk 6 above)
Comment (startcode) [reply third_party/adv_plat/include/bcan_defs.h]: please make this change for all other types

File: third_party/adv_plat/include/adv_trigger_ctl.h [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+* @author feiaiguo@, shuguoli@
+* @date 2016/07/11
+*
+* Baidu ADV (Autonomous Driving Vehicle) hardware trigger library, definitions.
+*
+* Modified by youxiangtao@
+* @date 2017/09/10
+**/
+
+#ifndef ADU_PLAT_SW_LIB_ADV_TRIGGER_ADV_TRIGGER_CTL_H
```
Comment (lianglia-apollo) [inline third_party/adv_plat/include/adv_trigger_ctl.h]: Please also change the guard to apollo rather than adu. Thanks!

**Pull request closed without merging**